econml/utilities.py

```diff
@@ -508,7 +508,8 @@ def check_inputs(Y, T, X, W=None, multi_output_T=True, multi_output_Y=True):
     X, T = check_X_y(X, T, multi_output=multi_output_T, y_numeric=True)
     _, Y = check_X_y(X, Y, multi_output=multi_output_Y, y_numeric=True)
     if W is not None:
-        W, _ = check_X_y(W, Y)
+        W, _ = check_X_y(W, Y, multi_output=multi_output_Y, y_numeric=True)
+        return Y, T, X, W
```

Comment:
Unneeded since we just do the same immediately following this `if`.
```suggestion
```

Reply:
Done